### PR TITLE
[IE TESTS] Add Cpu plugin as dependecies to the functional tests

### PR DESCRIPTION
**Pull request by iefode**
Add Multi, hetero and Cpu plugin as dependecies to the functional tests

### CODE REVIEW

File: inference-engine/tests/functional/plugin/cpu/CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,8 @@ addIeTargetTest(
         ROOT ${CMAKE_CURRENT_SOURCE_DIR}
         DEPENDENCIES
             MKLDNNPlugin
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/cpu/CMakeLists.txt:12]: why not to `funcSharedTests`?

Comment (iefode) [reply inference-engine/tests/functional/plugin/cpu/CMakeLists.txt:12]: Correct, Thank a lot. Fixed

File: inference-engine/tests/functional/plugin/shared/CMakeLists.txt [hunk 2]
```diff
@@ -8,6 +8,9 @@ set(TARGET_NAME funcSharedTests)
 list(APPEND EXPORT_DEPENDENCIES
         funcTestUtils
         ngraphFunctions
+        MKLDNNPlugin
+        MultiDevicePlugin
+        HeteroPlugin
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/CMakeLists.txt:13]: it's not export dependencies. Need to add to `addIeTarget` to the section `DEPENDENCIES`

Comment (iefode) [reply inference-engine/tests/functional/plugin/shared/CMakeLists.txt]: Done

File: inference-engine/tests/functional/plugin/shared/CMakeLists.txt [hunk 3]
```diff
@@ -23,6 +23,7 @@ addIeTarget(
         DEPENDENCIES
             HeteroPlugin
             MultiDevicePlugin
+            MKLDNNPlugin
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/plugin/shared/CMakeLists.txt:26]: looks like it does not work for ARM. Let's use:
```cmake
if(TARGET MKLDNNPlugin)
    add_dependencies(${TARGET_NAME} MKLDNNPlugin)
endif()
```

File: inference-engine/tests/functional/plugin/shared/CMakeLists.txt [hunk 4]
```diff
@@ -27,6 +27,10 @@ addIeTarget(
             ${EXPORT_DEPENDENCIES}
 )
 
+if (TARGET MKLDNNPlugin)
+    add_dependencies(${TARGET_NAME} MKLDNNPlugin)
+endif()
```
Comment (gladilov-gleb) [inline inference-engine/tests/functional/plugin/shared/CMakeLists.txt:32]: Is it okay to add as dependency MKLDNNPlugin only, while PR name is about multi and heter as well?

**Pull request merged**